### PR TITLE
added docker-build + cleanup

### PR DESCRIPTION
**Pull request by pdmthorsrud**
@fredrikhgrelland not sure if this is a help or contributing to the chaos, let me know!

### CODE REVIEW

File: template/docker/README.md [hunk 1]
```diff
@@ -1,2 +1,7 @@
 # Docker build directory
-TODO: write doc
\ No newline at end of file
+Put your Dockerfile and other files relating to a docker-build here. 
+### Building docker image locally
+`make build-docker [IMAGE_NAME=<name>:<tag>]` will look for a Dockerfile in [docker/](docker/) and build an image tagged with `IMAGE_NAME`. If `IMAGE_NAME` is not supplied it will be tagged `template:latest`.
```
Comment (fredrikhgrelland) [inline template/docker/README.md:4]:
```suggestion
If you have docker installed on your machine, you may `cd docker; docker build -t my_image:local .`  and build the image.
```

File: template/docker/README.md [hunk 2]
```diff
@@ -1,2 +1,7 @@
...
+
+### Building and testing the docker-image within the vagrant-hashistack box
+We advise you to build and test your docker image within the hashistack eco-system. Running `make test` will launch the [default playbook](../dev/ansible/playbook.yml) inside the box, and [test_example/](../test_example/) shows a simple build process for building and running the docker image using this. Refer to books in [test_example/dev/ansible](../test_example/dev/ansible) to see details.
```
Comment (fredrikhgrelland) [inline template/docker/README.md:7]: Se other comment, this should maybe be revised.

File: template/Makefile [hunk 3]
```diff
@@ -7,11 +7,16 @@ export PATH := $(shell pwd)/tmp:$(PATH)
...
 
-update-box:
-	@SSL_CERT_FILE=${SSL_CERT_FILE} CURL_CA_BUNDLE=${CURL_CA_BUNDLE} vagrant box update || (echo '\n\nIf you get an SSL error you might be behind a transparent proxy. \nMore info https://github.com/fredrikhgrelland/vagrant-hashistack/blob/master/README.md#if-you-are-behind-a-transparent-proxy\n\n' && exit 2)
+test: clean up
+
+example:
+	cd test_example; SSL_CERT_FILE=${SSL_CERT_FILE} CURL_CA_BUNDLE=${CURL_CA_BUNDLE} vagrant up --provision
+
+build-docker: check-dockerfile
+	cd docker; docker build -t $(IMAGE_NAME) .
```
[the remark this answers is not in the document]
Comment (pdmthorsrud) [reply template/Makefile:20]: When developing a new image I like to be able to build my image without having to worry about the box at all at first. Mainly to iron out small and stupid mistakes like syntax errors, wrong tag on image, etc. etc.. It does water down the "coerce developers to work they way we want" and "everything should be tested inside the box". I'm honestly not sure. How much do we trust others to actually do most of the testing inside the box?

File: template/docker/README.md (hunk 1 above)
Comment (pdmthorsrud) [reply template/docker/README.md:4]: This is if we remove `make build-docker` I assume? I like it more, let's do that

**Pull request merged**